cirq/ops/gate_operation.py

```diff
@@ -114,33 +116,57 @@ def _decompose_(self) -> 'cirq.OP_TREE':
                                                     NotImplemented)
 
     def _pauli_expansion_(self) -> value.LinearDict[str]:
-        return protocols.pauli_expansion(self.gate)
```

Comment:
Sure. The triggering event for me is that the `allow_decompose` parameter was not being respected in several places when GateOperation was in play. For example, `has_unitary(gate_op, allow_decompose=True)` would call `GateOperation._has_unitary_`, which would call `has_unitary(gate)` with no `allow_decompose=True`, resulting in the decomposition being computed.

Basically, if we bounce a protocol method back into itself then we can get an amplification effect where more work than necessary is done because fallback logic is run once per bounce-back-level. By delegating directly into the implementation of the sub gate, fallback logic is only run once. This makes things more efficient, and honestly a bit more stable or at least predictable.

Reply:
Well about decompose :).  We should probably make this the norm then throughout.